pkg/util/rand/rand.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rand provides utilities related to randomization.
+package rand
+
+import "math/rand"
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyz0123456789")
+
+// String generates a random alphanumeric string n characters long.
+func String(n int) string {
+	b := make([]rune, n)
```

Comment:
given that this is a utility function, we should guard against bad input like negative numbers.
If a negative number is passed, it will cause panic and program termination.
I think it would be more desirable to return (string, error) and we should use error if the input parameter is bad instead of just causing program termination.